internals/daemon/api_exec.go

```diff
@@ -73,13 +75,29 @@ func v1PostExec(c *Command, req *http.Request, _ *userState) Response {
 		return statusBadRequest("%v", err)
 	}
 
+	// Inherit the pebble daemon environment.
+	environment := make(map[string]string)
```

Comment:
This will work here, but I think we should do this one level down, in [`cmdstate.Exec`](https://github.com/canonical/pebble/blob/48d20dc71efb14ed52fdf4430c50f2b39a7676af/internals/overlord/cmdstate/request.go#L72), so that any caller that creates "exec" tasks gets the right behaviour, rather than just the API. I realize the API is the only user right now, but it seems the right place to put "business logic" there rather than in the API handler.

Reply:
Moved in https://github.com/canonical/pebble/pull/234/commits/00d7495b57e5532de07ecdfd9a49a3c53466f3d0. Thanks!